tests/testAvlModel.ml

```diff
@@ -32,7 +32,7 @@ let addr_gen = QCheck.Gen.(
   )
 
 let tez_gen = QCheck.Gen.(
-    map (fun x -> Ligo.tez_from_literal ((string_of_int x) ^ "mutez")) (0 -- max_int)
+    map (fun x -> Ligo.tez_from_literal ((string_of_int x) ^ "mutez")) (0 -- 10_000_000_000)
```

Comment:
This is a bit interesting.

If we're talking about some actual `tez`, this change is good, because I don't think there ever will be `2^64 mutez`).

But since there are places we get a `tez` parameter from outside, where they can possibly pass a really large number to us (eg. `checker.withdraw_tez((2**64, 1))`). It is not hard to imagine a situation where we do some calculations on it without checking for the overflow, then checking the now incorrect result against some number, and allowing the operation in.

So I think this is not a good _default_ for `tez_gen`, because it might give a false sense of security. We can always have another generator like `safe_tez_gen` which generates more realistic looking numbers.

Reply:
That's a great point! :ok_hand: I will go through all entrypoints that receive tez not necessarily reflecting real value and make sure we generate numbers in the entire spectrum for those.

However, we have several generators for tez already, depending on the usecase, and the `tez_gen` generator above is used _exclusively_ in `testAvlModel.ml` to represent actual collateral contained in slices. So, for this one I think we should avoid going up to `max_int` (the problem being of course that we add them up in the avl tree, and we get trivial overflows that are impossible in reality).

What do you think?

Reply:
Oh yeah, I do agree with that; when testing AVL there's no need to have unrealistic liquidation slices. My suggestion was only to rename the existing one to `safe_tez_gen` or something, so we won't forget about this caveat in future :).

Reply:
Ah, I see. Will do!

By the way, about

> But since there are places we get a `tez` parameter from outside, where they can possibly pass a really large number to us (eg. `checker.withdraw_tez((2**64, 1))`). It is not hard to imagine a situation where we do some calculations on it without checking for the overflow, then checking the now incorrect result against some number, and allowing the operation in.

I am thinking that is not possible for this to happen, since tezos fails on overflows (now we'll do too). Even if the user passes an unrealistic value and there is some overflow, the whole operation will fail instead. :thinking:

Reply:
Oh, that's a great default. Yeah, this changes things quite a bit, we might even disregard my previous comment then :). No need to be scared of overflows if they abort th execution.

Reply:
Great, I'll go ahead and mark this as "resolved" then! I went ahead and improved the names and added a couple of comments in 7d7acc3 anyway; some extra clarity is always welcome :slightly_smiling_face: